src/theory/arith/nl/nonlinear_extension.cpp

```diff
@@ -388,6 +390,19 @@ void NonlinearExtension::getAssertions(std::vector<Node>& assertions)
     }
   }
 
+  // Try to be "more deterministic" by adding assertions in the order they were
+  // given
+  for (auto it = d_containing.facts_begin(); it != d_containing.facts_end();
+       ++it)
+  {
+    Node lit = (*it).d_assertion;
+    auto iait = init_assertions.find(lit);
+    if (iait != init_assertions.end())
+    {
+      assertions.push_back(lit);
+      init_assertions.erase(iait);
+    }
+  }
```

Comment:
Is the block of code for adding assertions directly from init_assertions below necessary anymore?

Reply:
Unfortunately yes. The block above the new code may add new assertions to `init_assertions`...

Reply:
I see, so below we are adding the leftovers?  Can you perhaps add a comment above the block of code below that mentions this?